### PR TITLE
[WIP] Adds unit test for deployment secret removal

### PR DESCRIPTION
**Pull request by DhritiShikhar**
** THIS PR is on hold for next sprint**

### Motivation

Adds a test case for [this](https://github.com/redhat-developer/service-binding-operator/issues/422#issuecomment-618484986) scenario

### Changes

Adds a unit test for the scenario when secret is removed from deployment, the operator should reconcile to inject the secret again.

### CODE REVIEW

**Comment by Avni-Sharma** [comment]
Steps followed on cluster:
- Imported an app
- created a db instance
- created sbr cr
Checked deployment  `oc get deployments nodejs-rest-http-crud-git -o yaml -n=service-binding-demo`
the secret was in envFrom . Edited the deployment to delete the injected secret. Reconciliation does not happen. Now if I check the injected secret is not present

File: pkg/controller/servicebindingrequest/reconciler_test.go [hunk 1]
```diff
@@ -359,3 +359,59 @@ func TestEmptyApplicationSelector(t *testing.T) {
...
+	// update Application
+	d := appsv1.Deployment{}
+	require.NoError(t, fakeClient.Get(ctx, namespacedName, &d))
+	d.Spec.Template.Spec.Containers[0].EnvFrom[0].SecretRef.Name = ""
+	require.NoError(t, fakeClient.Update(ctx, &d))
+
+	reconciler = &Reconciler{client: fakeClient, dynClient: f.FakeDynClient(), scheme: f.S}
+	res, err = reconciler.Reconcile(reconcileRequest())
+	require.NoError(t, err)
+	require.False(t, res.Requeue)
```
Comment (Avni-Sharma) [inline pkg/controller/servicebindingrequest/reconciler_test.go:412]: wondering if require.False will help us in requeuing

Comment (baijum) [reply pkg/controller/servicebindingrequest/reconciler_test.go:412]: @Avni-Sharma Are you expecting Requeue True here?

Comment (Avni-Sharma) [reply pkg/controller/servicebindingrequest/reconciler_test.go:412]: I tried to run this on the cluster, but it is not working as the expected test. The secret is not getting injected again.

File: pkg/controller/servicebindingrequest/reconciler_test.go [hunk 2]
```diff
@@ -359,3 +359,59 @@ func TestEmptyApplicationSelector(t *testing.T) {
...
+	}
+	require.True(t, reflect.DeepEqual(expectedStatus, sbrOutput.Status.Applications[0]))
+
+	// update Application
+	d := appsv1.Deployment{}
+	require.NoError(t, fakeClient.Get(ctx, namespacedName, &d))
+	d.Spec.Template.Spec.Containers[0].EnvFrom[0].SecretRef.Name = ""
+	require.NoError(t, fakeClient.Update(ctx, &d))
+
+	reconciler = &Reconciler{client: fakeClient, dynClient: f.FakeDynClient(), scheme: f.S}
```
Comment (pedjak) [inline pkg/controller/servicebindingrequest/reconciler_test.go:409]: do we need to create a fresh instance, or we could reuse already created one?

**Comment by pedjak** [comment]
closing in favour of adding equivalent acceptance test

**Pull request closed without merging**